third_party/move/move-model/bytecode/abstract_domain_derive/tests/test.rs

```diff
@@ -0,0 +1,79 @@
+use abstract_domain_derive::AbstractDomain;
+use move_stackless_bytecode::dataflow_domains::{AbstractDomain, JoinResult};
+
+#[derive(AbstractDomain)]
+struct Unit;
+
+/// Add a formal top and bottom element to type `T`
+#[derive(PartialEq, Eq, Clone)]
+pub enum Plus2<T> {
```

Comment:
As you saved it here, perhaps add a comment that it might be useful elsewhere:

`/// Introduce struct for testing <OBJECTIVE>, perhaps will be also useful elsewhere`